archunit/src/main/java/com/tngtech/archunit/library/GeneralCodingRules.java

```diff
@@ -151,4 +154,28 @@ private GeneralCodingRules() {
     @PublicAPI(usage = ACCESS)
     public static final ArchRule NO_CLASSES_SHOULD_USE_JODATIME =
             noClasses().should(USE_JODATIME).because("modern Java projects use the [java.time] API instead");
+
+    /**
+     * For information about checking this condition, refer to {@link GeneralCodingRules}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public static final ArchCondition<JavaField> USE_FIELD_INJECTION = useFieldInjection().as("use field injection");
+
+    private static ArchCondition<JavaField> useFieldInjection() {
+        ArchCondition<JavaField> annotatedWithAutowired = beAnnotatedWith("org.springframework.beans.factory.annotation.Autowired");
+        ArchCondition<JavaField> annotatedWithValue = beAnnotatedWith("org.springframework.beans.factory.annotation.Value");
```

Comment:
I'm afraid it would be a bit inconvenient, as some standard Spring properties don't have corresponding classes annotated by `@ConfigurationProperties`. For example: `spring.application.name` - this property is usually injected by `@Value`

Reply:
`@Value` can be used on constructor parameters:
```java
public InfoBean(@Value("spring.application.name") String applicationName) {
  // ...
}
```

So there is no need to use `@Value` on fields.

Reply:
Oh... I see... I'm OK with it.